src/main/java/com/linkedin/kmf/common/Utils.java

```diff
@@ -108,6 +108,33 @@ public static int createMonitoringTopicIfNotExists(String zkUrl, String topic, i
     }
   }
 
+  /**
+   * Create general topic (for Kafka metrics reporter).
+   *
+   * @param zkUrl zookeeper connection url
+   * @param topic topic name
+   * @param replicationFactor replication factor
+   * @param partitionCount topic partition number
+   * @param topicConfig additional topic configuration
+   */
+  public static void createTopicIfNotExists(String zkUrl, String topic, int replicationFactor,
```

Comment:
Can we reduce the code duplication by reusing the existing `createTopicIfNotExists`? We can either just set partitionToBrokerRatio to be very small fraction number, e.g. 0.00001, or add a new parameter such as minPartitionNum. Then the topic can be created with `createTopicIfNotExists(partitionToBrokerRatio=0, minPartitionNum=1)`. What do you think?

Reply:
Thanks your review! Can we set the `createTopicIfNotExists` as primitive method which can be invoke by `createMonitoringTopicIfNotExists`  with `partitionToBrokerRatio` parameter?
Such as:
```
createMonitoringTopicIfNotExists(topic, partitionToBrokerRatio, ...) {
  createTopicIfNotExists(topic, partitionCount, factor, ...)
}
```
Maybe we just need a more general method `createTopicIfNotExists` rather than `createMonitoringTopicIfNotExists` ?

Reply:
Translating `partitionNum` to `partitionToBrokerRatio` requires your code to figure out the broker number and the code will be a little more complex. Yes, I think we can have a more general method createTopicIfNotExists(...) which takes parameters minPartitionToBrokerRatio and minPartitionNum. The partition number of the topic can be determined as max(minPartitionToBrokerRatio * brokerNum, minPartitionNum). Does this sound OK?

Reply:
Thanks your reply! So, I'll modify the method `createMonitoringTopicIfNotExists(zkUrl, topic, replicationFactor,partitionToBrokerRatio, topicConfig)` to the method `createTopicIfNotExists(zkUrl, topic, replicationFactor,partitionToBrokerRatio, minPartitionNum, topicConfig)`. is it reasonable for this?